File: app/src/main/java/com/lazylee/lzywanandroid/ui/adapter/ProjectFragmentAdapter.java
package com.lazylee.lzywanandroid.ui.adapter;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.lazylee.lzywanandroid.ui.activity.main.project.fragment.ProjectVPFragment;
import com.lazylee.lzywanandroid.data.entity.ProjectCategory;

import java.util.ArrayList;


public class ProjectFragmentAdapter extends FragmentPagerAdapter {

    private ArrayList<ProjectVPFragment> fragments;
    private ArrayList<ProjectCategory> categories;

    public ProjectFragmentAdapter(FragmentManager fm) {
        super(fm);
    }

    public ProjectFragmentAdapter(FragmentManager fm, ArrayList<ProjectVPFragment> fragments,
                                  ArrayList<ProjectCategory> categories) {
        super(fm);
        this.fragments = fragments;
        this.categories = categories;
    }

    @Override
    public Fragment getItem(int position) {
        return fragments.get(position);
    }

    @Override
    public int getCount() {
        return fragments == null ? 0 : categories.size();
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return categories.get(position).getName();
    }
}

File: app/src/main/java/com/lazylee/lzywanandroid/ui/adapter/ProjectFragmentAdapter.kt
package com.lazylee.lzywanandroid.ui.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

import com.lazylee.lzywanandroid.ui.activity.main.project.fragment.ProjectVPFragment
import com.lazylee.lzywanandroid.data.entity.ProjectCategory

import java.util.ArrayList


class ProjectFragmentAdapter : FragmentPagerAdapter {

    private val fragments: ArrayList<ProjectVPFragment>?
    private val categories: ArrayList<ProjectCategory>

    constructor(fm: FragmentManager) : super(fm) {}

    constructor(fm: FragmentManager, fragments: ArrayList<ProjectVPFragment>,
                categories: ArrayList<ProjectCategory>) : super(fm) {
        this.fragments = fragments
        this.categories = categories
    }

    override fun getItem(position: Int): Fragment {
        return fragments!![position]
    }

    override fun getCount(): Int {
        return if (fragments == null) 0 else categories.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return categories[position].name
    }
}
